tutor/templates/jobs/init/mounted-edx-platform.sh

```diff
@@ -0,0 +1,26 @@
+# When a new local copy of edx-platform is bind-mounted, certain build
+# artifacts from the openedx image's edx-platform directory are lost.
+# We regenerate them here.
+
+if [ -f /openedx/edx-platform/bindmount-canary ] ; then
+	# If this file exists, then edx-platform has not been bind-mounted,
+	# so no build artifacts need to be regenerated.
+	echo "Using edx-platform from image (not bind-mount)."
+	echo "No extra setup is required."
+	exit
+fi
+
+echo "Performing additional setup for bind-mounted edx-platform."
+set -x # Echo out executed lines
+
+# Regenerate Open_edX.egg-info
+pip install -e .
+
+# Regenerate node_modules
+npm clean-install
+
+# Regenerate static assets.
+openedx-assets build --env=dev
```

Comment:
These latter two steps will be performed every time we run `launch` with a bind-mount. I don't have a better alternative, so let's keep it that way, but we should keep it in mind.

Reply:
Something we could do, eventually, would be to use a hash to ensure that these steps are not run extraneously.
* node_modules would be easy: upon running `npm install`, create a node_modules/.package_lock_hash file, containing a hash of package-lock.json. If package-lock.json matches node_modules/.package_lock_hash, then we can skip node_modules installation.
* assets would be more complicated, but still doable. The scripts/build-assets.sh has the means to a compile a list of all asset input files. Saving the combined hash of those somewhere would allow us to skip asset compilation when assets are unchanged.

Thing is, this is just reimplementing what Docker already does. That's why I'd rather move [node_modules](https://github.com/openedx/wg-developer-experience/issues/150) and [assets](https://github.com/openedx/wg-developer-experience/issues/151) outside of the bind-mount if at all possible.